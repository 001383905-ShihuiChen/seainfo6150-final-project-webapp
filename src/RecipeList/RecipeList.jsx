import React from 'react'
import { Link } from "react-router-dom";
import PropTypes from "prop-types"
import RecipeItem from "../RecipeItem/RecipeItem.jsx";
import styles from "./RecipeList.module.css";
import Header from "../Header/Header";
const RecipeList = props => {
  return (
    <>
    <Header/>
    <div className={styles.wholeDiv}>
        
      <div className={styles.titleDiv}><h1>Let's find your own recipe!</h1></div>
      <ul className={styles.recipeList}>
        {props.recipe.map((recipe,index) => (
          <Link key={index} className={styles.clickBlock} to={"/recipe/"+recipe.categoryId+"/"+recipe.recipeId}>
            <li key={index}>
              <RecipeItem url={recipe.image._url} title={recipe.title} />
            </li>
          </Link>
        ))}
      </ul>
    </div>
    </>
  );
};

RecipeList.propTypes = {
  recipe: PropTypes.array.isRequired
};
export default RecipeList;