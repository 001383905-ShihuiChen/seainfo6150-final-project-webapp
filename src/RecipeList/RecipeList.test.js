import React from "react";
import { render } from "@testing-library/react";
import RecipeList from "./RecipeList.jsx";
import { BrowserRouter } from 'react-router-dom'

it("renders the RecipeList component correctly", () => {
  const test = [
    {
      categoryId: "breakfast",
      recipeId: "001-h",
      title: "Ham and Cheese Breakfast Tortillas",
      image: {
        _url: "https://imagesvc.meredithcorp.io/v3/mm/image?url=https%3A%2F%2Fimages.media-allrecipes.com%2Fuserphotos%2F4536480.jpg&w=596&h=596&c=sc&poi=face&q=85"
      },
    },
    {
        categoryId: "breakfast",
        recipeId: "002-q",
        title: "Quick & Crispy Home Fries",
        image: {
          _url: "https://imagesvc.meredithcorp.io/v3/mm/image?url=https%3A%2F%2Fimages.media-allrecipes.com%2Fuserphotos%2F6991200.jpg&w=596&h=596&c=sc&poi=face&q=85"
        },
    }
  ]

  // For component with <Link> must nest within <Router>
  const { container } = render(
    <BrowserRouter>
      <RecipeList
        recipe={Object.values(test)}
        pageTitle="category"
      />
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});