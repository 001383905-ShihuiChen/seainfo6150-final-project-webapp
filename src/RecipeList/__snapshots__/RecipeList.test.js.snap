// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the RecipeList component correctly 1`] = `
<div>
  <div>
    <h1
      class="container"
    >
      Wonderful Recipe
    </h1>
    <nav>
      <ul
        class="list"
      >
        <li
          class="navContainer"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="navContainer"
        >
          <a
            href="/category"
          >
            Category
          </a>
        </li>
        <li
          class="navContainer"
        >
          <a
            href="/aboutus"
          >
            About Us
          </a>
        </li>
        <li
          class="navContainer"
        >
          <a
            href="/contactus"
          >
            Contact Us
          </a>
        </li>
        <li
          class="navContainer"
        >
          <a
            href="/login"
          >
            Log In
          </a>
        </li>
      </ul>
    </nav>
    <br />
  </div>
  <div
    class="wholeDiv"
  >
    <div
      class="titleDiv"
    >
      <h1>
        Let's find your own recipe!
      </h1>
    </div>
    <ul
      class="recipeList"
    >
      <a
        class="clickBlock"
        href="/recipe/breakfast/001-h"
      >
        <li>
          <div>
            <div
              class="imageStyle"
            >
              <img
                alt="Ham and Cheese Breakfast Tortillas"
                src="https://imagesvc.meredithcorp.io/v3/mm/image?url=https%3A%2F%2Fimages.media-allrecipes.com%2Fuserphotos%2F4536480.jpg&w=596&h=596&c=sc&poi=face&q=85"
              />
            </div>
            <p>
              Ham and Cheese Breakfast Tortillas
            </p>
          </div>
        </li>
      </a>
      <a
        class="clickBlock"
        href="/recipe/breakfast/002-q"
      >
        <li>
          <div>
            <div
              class="imageStyle"
            >
              <img
                alt="Quick & Crispy Home Fries"
                src="https://imagesvc.meredithcorp.io/v3/mm/image?url=https%3A%2F%2Fimages.media-allrecipes.com%2Fuserphotos%2F6991200.jpg&w=596&h=596&c=sc&poi=face&q=85"
              />
            </div>
            <p>
              Quick & Crispy Home Fries
            </p>
          </div>
        </li>
      </a>
    </ul>
  </div>
</div>
`;
